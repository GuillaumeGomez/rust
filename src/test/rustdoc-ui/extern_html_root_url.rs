#![crate_type = "lib"]

#![doc(extern_html_root_url)]
//~^ ERROR
#![doc(extern_html_root_url = "a")]
//~^ ERROR
#![doc(extern_html_root_url(a))]
//~^ ERROR
#![doc(extern_html_root_url(a()))]
//~^ ERROR
#![doc(extern_html_root_url(a = 1))]
//~^ ERROR
#![doc(extern_html_root_url(a = ""))]
//~^ ERROR
#![doc(extern_html_root_url(a = "1"))] // This one is supposed to work.
#![doc(extern_html_root_url(b = "2", c = "3"))] // This one is supposed to work.

pub fn foo() {}
