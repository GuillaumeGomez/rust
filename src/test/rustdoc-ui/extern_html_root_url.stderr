error: extern_html_root_url only accepts this form: `extern_html_root_url(crate = "url")`
  --> $DIR/extern_html_root_url.rs:3:8
   |
LL | #![doc(extern_html_root_url)]
   |        ^^^^^^^^^^^^^^^^^^^^

error: extern_html_root_url only accepts this form: `extern_html_root_url(crate = "url")`
  --> $DIR/extern_html_root_url.rs:5:8
   |
LL | #![doc(extern_html_root_url = "a")]
   |        ^^^^^^^^^^^^^^^^^^^^^^^^^^

error: extern_html_root_url() only accepts `crate_name = "url"`
  --> $DIR/extern_html_root_url.rs:7:29
   |
LL | #![doc(extern_html_root_url(a))]
   |                             ^

error: extern_html_root_url() only accepts `crate_name = "url"`
  --> $DIR/extern_html_root_url.rs:9:29
   |
LL | #![doc(extern_html_root_url(a()))]
   |                             ^^^

error: extern_html_root_url() only accepts `crate_name = "url"`
  --> $DIR/extern_html_root_url.rs:11:29
   |
LL | #![doc(extern_html_root_url(a = 1))]
   |                             ^^^^^

error: URL cannot be empty
  --> $DIR/extern_html_root_url.rs:13:29
   |
LL | #![doc(extern_html_root_url(a = ""))]
   |                             ^^^^^^

error: aborting due to 6 previous errors

